ceno_zkvm/examples/riscv_opcodes.rs

```diff
@@ -329,4 +339,5 @@ fn main() {
             }
         };
     }
+    exit_span!(top_level);
```

Comment:
Btw, what happens if you forget to exit a span, or exit them in the wrong order?

Reply:
> Btw, what happens if you forget to exit a span, or exit them in the wrong order?

Right now it seems to misrecord the offending spans silently :). I think in general it depends on the implementation of the subscriber/layer that processes them. A custom one could panic instead. 

> Might be useful to have a macro that works like this:

I understand the benefit, but for me the extra nesting makes the code less readable. Also the `enter/exit` pattern existed already in the codebase.

Reply:
Or they could be a check a the end of the program that all spans were closed. If that's possible to know.

Reply:
If you forget the `exit_span` you get a word from the compiler (should be refused by CI):

`warning: unused variable: my_span`

and if you forget the `let my_span`:

`warning: unused `EnteredSpan` that must be used`